src/analysis/normed_space/operator_norm.lean

```diff
@@ -404,6 +412,9 @@ lemma op_norm_comp_le (f : E →SL[σ₁₂] F) : ∥h.comp f∥ ≤ ∥h∥ * 
 (cInf_le bounds_bdd_below
   ⟨mul_nonneg (op_norm_nonneg _) (op_norm_nonneg _), λ x,
     by { rw mul_assoc, exact h.le_op_norm_of_le (f.le_op_norm x) } ⟩)
+
+lemma op_nnnorm_comp_le [ring_hom_isometric σ₁₃] (f : E →SL[σ₁₂] F) : ∥h.comp f∥₊ ≤ ∥h∥₊ * ∥f∥₊ :=
```

Comment:
How come we need the extra `[ring_hom_isometric σ₁₃]` here?

Reply:
It's because we need the `semi_normed_group` instance to state this lemma (as that's where `has_nnnorm` comes from), while we only needed the `has_norm` instance to state the lemma above.